legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/util/LogicalPlanUtils.java

```diff
@@ -282,4 +314,22 @@ public static Selection getRowsBasedOnLatestTimestamp(Dataset dataset, String fi
 
         return Selection.builder().source(dataset.datasetReference()).condition(condition).addFields(countFunction).build();
     }
+
+    public static void validateOptimizationFilters(List<OptimizationFilter> optimizationFilters, Dataset dataset)
```

Comment:
Not critical - may consider putting the validation method in the relevant Planner class itself, which is what we have been doing, for consistency